sonic-chassisd/scripts/chassisd

```diff
@@ -208,9 +208,10 @@ class ModuleUpdater(logger.Logger):
             self.chassis_table._del(CHASSIS_INFO_KEY_TEMPLATE.format(1))
 
         if self.asic_table is not None:
-            asics = list(self.asic_table.getKeys())
-            for asic in asics:
-                self.asic_table._del(asic)
+            if not self._is_supervisor():
```

Comment:
can we debug why PMON is down at the first place? Can we ensure PMON is running? Can we modify the test to ensure PMON is running?

Reply:
> can we debug why PMON is down at the first place? -> this is done by the test case.
>Can we ensure PMON is running? -> that will be a change in the test case.
>Can we modify the test to ensure PMON is running? -> this will change the behavior of the test case. 

This PR is intended to keep the Fabric entries in STATE_DB when PMON is stopped in supervisor by the test case so syncd and swss will remain up state. Regardless of changes required in test case, this code change is required to make sure that the syncd and swss are up and running in case pmon is crashed/shutdown.

Reply:
@prgeor , bringing the pmon down was part of the test case. Test case failed because container checker could not find any asic entry to report for corresponding syncd/swss docker down.

@prem-nokia , Thanks for the fix! During the init, are these entries recreated from fresh in case any asic is removed due to physical FC removal?

Reply:
> 
@anamehra  The Entry will be recreated when PMON is restarted.  We have tested it and works as expected

Reply:
@mlok-nokia can you point to code where this happens? If i understand correctly issue was once we do de-init we were losing the entry and not getting repopulated.

Reply:
Before this PR, these "FABRIC_ASIC" entries  will be removed if stop chassid or PMON (which trigger the de_init).   The entries will be recreated when chassisd or PMON is restarted

Reply:
Hi @abdosi , in case of a failure scenario, the pmon docker stays down (as per testcase) and that is why init was not called again. So the plan was to not clear the entries in de-init path so that the container_checker could still access the FC asic entries even if the chassisd is down.